rotkehlchen/chain/ethereum/interfaces/ammswap/ammswap.py

```diff
@@ -0,0 +1,991 @@
+                            address=address,
+                            from_address=from_address_deserialized,
+                            to_address=to_address_deserialized,
+                            timestamp=Timestamp(int(timestamp)),
+                            location=self.location,
+                            token0=token0,
+                            token1=token1,
+                            amount0_in=AssetAmount(amount0_in),
+                            amount1_in=AssetAmount(amount1_in),
+                            amount0_out=AssetAmount(amount0_out),
+                            amount1_out=AssetAmount(amount1_out),
+                        ))
+                except KeyError as e:
+                    log.error(
+                        f'Failed to read trade in {self.location} swap {str(entry)}. '
+                        f'{str(e)}.',
+                    )
+                    continue
+
+                # with the new logic the list of swaps can be empty, in that case don't try
+                # to make trades from the swaps
+                if len(swaps) == 0:
+                    continue
+
+                # Now that we got all swaps for a transaction, create the trade object
+                trades.extend(self._tx_swaps_to_trades(swaps))
+
+            # Check whether an extra request is needed
+            if len(result['swaps']) < GRAPH_QUERY_LIMIT:
+                break
+
+            # Update pagination step
+            param_values = {
+                **param_values,
+                'offset': param_values['offset'] + GRAPH_QUERY_LIMIT,  # type: ignore
+            }
+
+        return trades
+
+    def _get_trades_graph(
+            self,
+            addresses: List[ChecksumEthAddress],
+            start_ts: Timestamp,
+            end_ts: Timestamp,
+    ) -> AddressTrades:
+        address_trades = {}
+        for address in addresses:
+            trades = self._get_trades_graph_for_address(address, start_ts, end_ts)
+            if len(trades) != 0:
+                address_trades[address] = trades
+
+        return address_trades
+
+    def _get_trades(
+            self,
+            addresses: List[ChecksumEthAddress],
+            from_timestamp: Timestamp,
+            to_timestamp: Timestamp,
+            only_cache: bool,
+    ) -> AddressTrades:
+        """Request via graph all trades for new addresses and the latest ones
+        for already existing addresses. Then the requested trade are written in
+        DB and finally all DB trades are read and returned.
+        """
+        address_amm_trades: AddressTrades = {}
+        new_addresses: List[ChecksumEthAddress] = []
+        existing_addresses: List[ChecksumEthAddress] = []
+        min_end_ts: Timestamp = to_timestamp
+
+        if only_cache:
+            return self._fetch_trades_from_db(addresses, from_timestamp, to_timestamp)
+
+        # Get addresses' last used query range for Sushiswap trades
```

Comment:
Comment references sushiswap

```suggestion
        # Get addresses' last used query range for this AMM's trades
```